src/Components/ADT3DSceneBuilder/Internal/VisualRuleForm/VisualRuleForm.tsx

```diff
@@ -0,0 +1,238 @@
+    IVisualRuleFormStylesProps
+} from './VisualRuleForm.types';
+
+const getClassNames = classNamesFunction<
+    IVisualRuleFormStylesProps,
+    IVisualRuleFormStyles
+>();
+
+const VisualRuleForm: React.FC<IVisualRuleFormProps> = (props) => {
+    // Props
+    const {
+        isPropertyTypeDropdownEnabled,
+        rootHeight,
+        setPropertyTypeDropdownEnabled,
+        styles
+    } = props;
+
+    // General constants
+    const { t } = useTranslation();
+    const theme = useTheme();
+    const classNames = getClassNames(styles, {
+        theme: theme
+    });
+    const commonFormStyles = getPanelFormStyles(theme, rootHeight);
+    const typeOptions: Array<IDropdownOption> = useMemo(
+        () =>
+            Object.keys(DTDLPropertyIconographyMap).map((mappingKey) => ({
+                key: `value-type-${DTDLPropertyIconographyMap[mappingKey].text}`,
+                text: DTDLPropertyIconographyMap[mappingKey].text,
+                data: { icon: DTDLPropertyIconographyMap[mappingKey].icon }
+            })),
+        []
+    );
+    // TODO: Wire this up to actual form data
+    const formData = {
+        propertyType: 'boolean'
+    };
+
+    // Contexts
+    const {
+        config,
+        sceneId,
+        adapter,
+        setVisualRuleFormMode,
+        state: { selectedElements, selectedBehavior }
+    } = useContext(SceneBuilderContext);
+
+    // TODO: Update form callbacks to wire them to form state
+    const onTabValidityChange = useCallback(
+        (_tabName: TabNames, _state: IValidityState) => {
+            return;
+        },
+        []
+    );
+
+    const onPropertyChange = useCallback(
+        (_newPropertyExpression: PropertyExpression) => {
+            return;
+        },
+        []
+    );
+
+    const onTypeChange = useCallback(
+        (
+            _event: React.FormEvent<HTMLDivElement>,
+            _option?: IDropdownOption<any>,
+            _index?: number
+        ) => {
+            return;
+        },
+        []
+    );
+
+    const customStyles = mergeStyleSets({
+        description: {
+            fontSize: FontSizes.size14,
+            color: theme.palette.neutralSecondary
+        }
+    });
+
+    const onInternalModeChanged = useCallback((internalMode) => {
+        if (internalMode === 'INTELLISENSE') {
+            setPropertyTypeDropdownEnabled(true);
+        } else {
+            setPropertyTypeDropdownEnabled(false);
+        }
+    }, []);
+
+    return (
+        <>
+            <div className={commonFormStyles.content}>
+                <div className={commonFormStyles.header}>
+                    <Stack tokens={{ childrenGap: 12 }}>
+                        <div className={customStyles.description}>
+                            {t('3dSceneBuilder.visualRuleForm.formDescription')}
+                        </div>
+                        <TextField
+                            label={t('displayName')}
+                            // value={FormState}
```

Comment:
remove?

Reply:
Comments in this file are placeholders to remember where to have certain props or values, will remove in next PR